golos.publication/golos.publication.cpp

```diff
@@ -325,10 +336,10 @@ void publication::close_message(account_name account, uint64_t id) {
         pools.modify(pool, _self, [&](auto &item) { item.state = state; });
 }
 
-void publication::close_message_timer(account_name account, uint64_t id) {
+void publication::close_message_timer(account_name account, uint64_t id, uint64_t seconds_diff) {
     transaction trx;
     trx.actions.emplace_back(action{permission_level(_self, N(active)), _self, N(closemssg), structures::accandvalue{account, id}});
-    trx.delay_sec = config::cashout_window;
+    trx.delay_sec = config::cashout_window > seconds_diff ? config::cashout_window - seconds_diff : 0;
```

Comment:
1. comment, written to closed post is non-consensus comment, it have no rewards, etc. maybe we should have some lite version of close-function to just fix comment state
2. if delay == 0, there is no need to send transaction (and consume RAM), we can call inline action or even function within current action